client/ayon_core/tools/publisher/control.py

```diff
@@ -1827,8 +1838,13 @@ def _reset_plugins(self):
     def _collect_creator_items(self):
         # TODO add crashed initialization of create plugins to report
         output = {}
+        allowed_creator_labels = self._get_allowed_creator_labels()
         for identifier, creator in self._create_context.creators.items():
             try:
+                if (not self._is_label_allowed(
+                        creator.label, allowed_creator_labels)):
+                    self.log.debug(f"{creator.label} not allowed for context")
+                    continue
```

Comment:
You're still recompiling and joining the regexes in the for loop - which we don't want. Plus we're putting 'too much' logic in the try..except that may just fail without us wanting to ignore it.

We're better off preparing the pattern before the loop. Like this:
```suggestion
        allowed_creator_labels = self._get_allowed_creator_labels()
        allowed_creator_label_regex = re.compile("|".join(allowed_created_labels))
        for identifier, creator in self._create_context.creators.items():
            try:
                if not allowed_creator_label_regex.match(creator.label):
                    self.log.debug(f"{creator.label} not allowed for context")
                    continue
```
However, that wouldn't handle the no filters case - but similar to that we could make it `None`.  just make sure to `join` the list and pattern outside of the loop instead of doing this per creator.

Reply:
I think we are sacrificing encapsulation and readability to negligent speed improvement.

Reply:
The joining on each iteration will be relatively costly - likely slower even than your previous code with the regex match per pattern due to that caching internally per frame.

Also, the comparison should not be in the try except.


I feel like here we're not suffering that much readability actually
```python
allowed_creators_pattern = self._get_allowed_creators_pattern()
for identifier, creator in self._create_context.creators.items():
    if (
        allowed_creators_pattern is not None
        and not allowed_creators_pattern.match(creator.label)
    ):
       continue

    try:
        output[identifier] = CreatorItem.from_creator(creator)
    except Exception:
        self.log.error(
            "Failed to create creator item for '%s'",
            identifier,
            exc_info=True
        )
```

where `_get_allowed_creators_pattern()` returns `Optional[re.Pattern]`

Reply:
Updated